tests/ui/conftest.py

```diff
@@ -49,16 +88,166 @@ def jwt_secret(base_url, variables):
 
 
 @pytest.fixture
-def initial_data(transactional_db):
-    from olympia.amo.tests import addon_factory, user_factory
-    from olympia.addons.models import AddonUser
+def initial_data(transactional_db, pytestconfig):
+    """
+    Fixture used to fill database will dummy addons.
+
+    Creates exactly 10 random addons with users that are also randomly
+    generated.
+    """
+    if not pytestconfig.option.usingliveserver:
+        return
+    for _ in range(10):
+        AddonUser.objects.create(
+            user=user_factory(), addon=addon_factory())
+
+
+@pytest.fixture
+def theme(transactional_db, create_superuser, pytestconfig):
+    """
+    Creates a custom theme named 'Ui-Test Theme'. This theme will be a
+    featured theme and will belong to the user created by the
+    'create_superuser' fixture.
+
+    It has one author.
+    """
+    if not pytestconfig.option.usingliveserver:
+        return
+    addon = addon_factory(
+        status=STATUS_PUBLIC,
+        type=ADDON_PERSONA,
+        average_daily_users=4242,
+        users=[UserProfile.objects.get(username='uitest')],
+        average_rating=4.21,
+        description=u'My UI Theme description',
+        file_kw={
+            'hash': 'fakehash',
+            'platform': amo.PLATFORM_ALL.id,
+            'size': 42,
+        },
+        guid=generate_addon_guid(),
+        homepage=u'https://www.example.org/',
+        name=u'Ui-Test Theme',
+        public_stats=True,
+        slug='ui-test',
+        summary=u'My UI theme summary',
+        support_email=u'support@example.org',
+        support_url=u'https://support.example.org/support/ui-theme-addon/',
+        tags=['some_tag', 'another_tag', 'ui-testing',
+                'selenium', 'python'],
+        total_reviews=777,
+        weekly_downloads=123456,
+        developer_comments='This is a testing theme, used within pytest.',
+    )
+    addon.save()
+    generate_collection(addon, app=FIREFOX,
+                        author=UserProfile.objects.get(username='uitest'),
+                        )
 
-    for x in range(10):
-        AddonUser.objects.create(user=user_factory(), addon=addon_factory())
+    print('Created Theme {0} for testing successfully'.format(addon.name))
+    return addon
+
+
+@pytest.fixture
+def addon(transactional_db, create_superuser, pytestconfig):
+    """
+    Creates a custom addon named 'Ui-Addon'. This addon will be a featured
+    addon and will have a featured collecton attatched to it. It will
+    belong to the user created by the 'create_superuser' fixture.
+
+    It has 1 preview, 5 reviews, and 2 authors. The second author is named
+    'ui-tester2'. It has a version number as well as a beta version.
+    """
+
```

Comment:
not needed whitespace after function definition. Also, this is not done in every function so let's maybe not do it at all :)

I'd even go ahead and do something like

```python
def addon(...)
    """
    ...
    """
    if not pytestconfig.option.usingliveserver:
        return

    default_icons = ...
    ...
```

so that you have a proper skipped line to switch context between okay… this is the same everywhre and now comes the actual function body